config/derive/src/lib.rs

```diff
@@ -45,9 +45,9 @@ fn get_type_argument<'a, 'b>(s: &'a str, ty: &'b Type) -> Option<&'b GenericArgu
 
 fn is_arc_rwlock(ty: &Type) -> bool {
     let dearced_ty = get_type_argument("Arc", ty)
-        .and_then(|ty| {
-            if let GenericArgument::Type(ty) = ty {
-                Some(ty)
+        .and_then(|typ| {
```

Comment:
`ty` is a convention usually in macros for type. I would suggest to leave it as it is or use full name e.g. `type`.